gulpjs/Dockerfile

```diff
@@ -0,0 +1,13 @@
+# Use the base App Engine Docker image, based on debian jessie.
+FROM debian:8
+
+# Install updates and dependencies
+RUN apt-get update -y && apt-get install --no-install-recommends -y -q curl python build-essential git ca-certificates libkrb5-dev imagemagick && \
+    apt-get clean && rm /var/lib/apt/lists/*_*
+
+# Install the latest LTS release of nodejs
+RUN mkdir /nodejs && curl https://nodejs.org/dist/v6.9.1/node-v6.9.1-linux-x64.tar.gz | tar xvzf - -C /nodejs --strip-components=1
+ENV PATH $PATH:/nodejs/bin
+RUN npm install -g gulp-cli
```

Comment:
Agreed that maybe pinning it to head is not the best, we could switch to specific versions. 

Right now there are multiple ways in the community to trigger gulp builds within a project:
1. "make" routing to the node_module dependency command
2. npm script
3. running the script via a node file or command line

The goal that I was trying to make was to abstract the different use cases for developers so that they can simplify their build steps. For my teams we usually use #1 but that requires both a dev dependency install and a prod install(two steps). Abstracting the gulp from the build avoids the use of installing dev dependencies if you wanted to do something simple using gulp.